.vscode/settings.json

```diff
@@ -0,0 +1,3 @@
+{
+  "editor.defaultFormatter": "rust-lang.rust-analyzer"
```

Comment:
Overall I'm kinda not a fan of including any VSCode artifacts in the repo, might wanna include ```.vscode/*``` or whatever into the gitignore

Reply:
i think we should include this so that everyone thats using the codebase has the same formatter and such. I may be mistaken though